plonky3/src/stark.rs

```diff
@@ -279,33 +313,28 @@ mod tests {
 
         let pil = pipeline.compute_optimized_pil().unwrap();
         let witness_callback = pipeline.witgen_callback().unwrap();
-        let witness = pipeline.compute_witness().unwrap();
+        let witness = &mut pipeline.compute_witness().unwrap();
```

Comment:
Again, very surprised this compiles. I don't know what is happening here.

You are taking an `&mut` from inside an `Arc`, but that should not be possible, it doesn't implement `DerefMut`.

Reply:
Seems like the same explanation? Would it be possible if I had introduced an intermediate variable, and taken an &mut on that? Then Rust does that automatically in this particular case, apparently.